xilem/examples/variable_clock.rs

```diff
@@ -0,0 +1,221 @@
+use xilem::{
+    view::{button, flex, label, prose, sized_box, variable_label, Axis, FlexExt, FlexSpacer},
+    Color, EventLoop, EventLoopBuilder, WidgetView, Xilem,
+};
+
+/// The state of the application, owned by Xilem and updated by the callbacks below.
+struct Clocks {
+    /// The font [weight](Weight) used for the values.
+    weight: f32,
+    /// The current UTC offset on this machine.
+    local_offset: Result<UtcOffset, IndeterminateOffset>,
+    /// The current time.
+    now_utc: OffsetDateTime,
+}
+
+impl Clocks {
+    fn local_now(&self) -> OffsetDateTime {
+        match self.local_offset {
+            Ok(offset) => self.now_utc.to_offset(offset),
+            Err(_) => self.now_utc,
+        }
+    }
+}
+
+/// A possible timezone, with an offset from UTC.
+struct TimeZone {
+    /// An approximate region which this offset applies to.
+    region: &'static str,
+    /// The offset from UTC
+    offset: time::UtcOffset,
+}
+
+impl TimeZone {
+    fn view(&self, data: &mut Clocks) -> impl WidgetView<Clocks> {
+        let date_time_in_self = data.now_utc.to_offset(self.offset);
+        sized_box(flex((
+            flex((
+                prose(self.region),
+                FlexSpacer::Flex(1.),
+                label(format!("UTC{}", self.offset)).brush(
+                    if data.local_offset.is_ok_and(|it| it == self.offset) {
+                        Color::ORANGE
+                    } else {
+                        masonry::theme::TEXT_COLOR
+                    },
+                ),
+            ))
+            .must_fill_major_axis(true)
+            .direction(Axis::Horizontal)
+            .flex(1.),
+            flex((
+                variable_label(
+                    date_time_in_self
+                        .format(format_description!("[hour repr:24]:[minute]:[second]"))
+                        .unwrap()
+                        .to_string(),
+                )
+                .text_size(48.)
+                // Use the roboto flex we have just loaded.
+                .with_font(FontStack::List(&[FontFamily::Named("Roboto Flex")]))
+                .target_weight(data.weight, 400.),
+                FlexSpacer::Flex(1.0),
+                (data.local_now().date() != date_time_in_self.date()).then(|| {
+                    label(
+                        date_time_in_self
+                            .format(format_description!("([day] [month repr:short])"))
+                            .unwrap(),
+                    )
+                }),
+            ))
+            .direction(Axis::Horizontal),
+        )))
+        .expand_width()
+        .height(72.)
+    }
+}
+
+fn app_logic(data: &mut Clocks) -> impl WidgetView<Clocks> {
+    flex((
+        // HACK: We add a spacer at the top for Android. See https://github.com/rust-windowing/winit/issues/2308
+        FlexSpacer::Fixed(40.),
+        controls(),
+        TIMEZONES.iter().map(|it| it.view(data)).collect::<Vec<_>>(),
```

Comment:
I mean, not super important, but would it make sense to do this on 2 columns until we have a scroll view (to reduce the chance of this being cut-off)

Reply:
There are two sides to this:
1) This was my original plan
2) I think this version looks better on a "normal resolution" Android device, which is where this example is primarily targeted.